### PR TITLE
Patched changes from main repo

### PR DESCRIPTION
**Pull request by molguin92**
Server now keeps track of state and shares it with the client.

### CODE REVIEW

File: lego_server.py [hunk 1]
```diff
@@ -90,6 +90,10 @@ def _handle_img(self, img):
             result, img_guidance = self.task.get_first_guidance()
             zc.check_and_display('guidance', img_guidance, display_list, wait_time = config.DISPLAY_WAIT_TIME, resize_max = config.DISPLAY_MAX_PIXEL)
             self.is_first_frame = False
+
```
Comment (junjuew) [inline lego_server.py]: remove the empty line here?

File: lego_server.py [hunk 2]
```diff
@@ -90,6 +90,10 @@ def _handle_img(self, img):
...
+            # first step
+            result['state_index'] = 0
+
```
Comment (junjuew) [inline lego_server.py]: remove the empty line

File: lego_server.py [hunk 3]
```diff
@@ -142,6 +146,15 @@ def _handle_img(self, img):
             self.task.update_state(bitmap)
             result, img_guidance = self.task.get_guidance()
 
+            if self.task.is_final_state():
+                step_idx = len(self.task.states)
+            else:
+                # get current step
+                step_idx = self.task.state2idx(self.task.current_state)
+                step_idx = -1 if step_idx < 0 else step_idx + 1
+
```
Comment (junjuew) [inline lego_server.py]: remove the empty line

File: lego_server.py [hunk 4]
```diff
@@ -142,6 +146,15 @@ def _handle_img(self, img):
             self.task.update_state(bitmap)
             result, img_guidance = self.task.get_guidance()
 
+            if self.task.is_final_state():
+                step_idx = len(self.task.states)
+            else:
+                # get current step
+                step_idx = self.task.state2idx(self.task.current_state)
+                step_idx = -1 if step_idx < 0 else step_idx + 1
```
Comment (junjuew) [inline lego_server.py]: Some comments on what the meaning of -1 and step_idx + 1?

**Review by junjuew — CHANGES_REQUESTED** [review]
add in some style and comment suggestions

**Comment by molguin92** [comment]
I'll fix this later today!

/Manuel

On April 30, 2018 8:19:59 PM GMT+02:00, Junjue Wang <notifications@github.com> wrote:
>junjuew requested changes on this pull request.
>
>add in some style and comment suggestions
>
>> @@ -90,6 +90,10 @@ def _handle_img(self, img):
>             result, img_guidance = self.task.get_first_guidance()
>zc.check_and_display('guidance', img_guidance, display_list, wait_time
>= config.DISPLAY_WAIT_TIME, resize_max = config.DISPLAY_MAX_PIXEL)
>             self.is_first_frame = False
>+
>
>remove the empty line here?
>
>> @@ -90,6 +90,10 @@ def _handle_img(self, img):
>             result, img_guidance = self.task.get_first_guidance()
>zc.check_and_display('guidance', img_guidance, display_list, wait_time
>= config.DISPLAY_WAIT_TIME, resize_max = config.DISPLAY_MAX_PIXEL)
>             self.is_first_frame = False
>+
>+            # first step
>+            result['state_index'] = 0
>+
>
>remove the empty line
>
>> @@ -142,6 +146,15 @@ def _handle_img(self, img):
>             self.task.update_state(bitmap)
>             result, img_guidance = self.task.get_guidance()
> 
>+            if self.task.is_final_state():
>+                step_idx = len(self.task.states)
>+            else:
>+                # get current step
>+                step_idx =
>self.task.state2idx(self.task.current_state)
>+                step_idx = -1 if step_idx < 0 else step_idx + 1
>+
>
>remove the empty line
>
>> @@ -142,6 +146,15 @@ def _handle_img(self, img):
>             self.task.update_state(bitmap)
>             result, img_guidance = self.task.get_guidance()
> 
>+            if self.task.is_final_state():
>+                step_idx = len(self.task.states)
>+            else:
>+                # get current step
>+                step_idx =
>self.task.state2idx(self.task.current_state)
>+                step_idx = -1 if step_idx < 0 else step_idx + 1
>
>Some comments on what the meaning of -1 and step_idx + 1?

--

Manuel Osvaldo Olguín
Doctoral Student

KTH Royal Institute of Technology
School of Electrical Engineering and Computer Science
Department of Information Science and Engineering

Osquldas väg 10, SE-100 44 Stockholm, Sweden
Mobile: +46 73 652 76 28
molguin@kth.se, www.kth.se

**Comment by molguin92** [comment]
I've incorporated the requested changes.

**Review by junjuew — APPROVED** [review]

**Pull request merged**